### PR TITLE
adds specs for knative-serving-utils and refactors service

### PR DESCRIPTION
**Pull request by invincibleJai**
- adds specs for knative-serving-utils and refactors service

Tracks: https://jira.coreos.com/browse/ODC-1931

### CODE REVIEW

**Comment by andrewballantyne** [comment]
@invincibleJai per @rohitkrai03 's comment on the previous iteration (https://github.com/openshift/console/pull/2829#discussion_r328275366) please adjust the signature of the `getKnativeServiceDepResource` to be a single argument and let it pull the values it needs out of it.

Also, there appears to be duplicate code found in `frontend/packages/knative-plugin/src/utils/create-knative-utils.ts` - the elements returned from `knativeServingResources` are repeated in other utilities.

Could you refactor these to call each other and share the values? Code duplication should be as minimal as possible.

**Comment by invincibleJai** [comment]
> @invincibleJai per @rohitkrai03 's comment on the previous iteration (#2829 (comment)) please adjust the signature of the getKnativeServiceDepResource to be a single argument and let it pull the values it needs out of it.

i was looking into this, `getKnativeServiceDepResource` is a common utility which takes data from two different from "DeplyImage" and "importForm" hence have two different types `GitImportFormData`  and `DeployImageFormData`.  So will pass `formData` and pull in util but still there will be few extra params e.g git is there in import and not in deploy and imageStreamResponse.

File: frontend/packages/knative-plugin/src/utils/create-knative-utils.ts [hunk 1]
```diff
@@ -7,39 +7,27 @@ import {
...
-export const createKnativeService = (
-  name: string,
-  applicationName: string,
-  namespace: string,
-  scaling: ServerlessScaling,
-  limits: LimitsData,
-  unknownTargetPort,
-  labels,
+export const getKnativeServiceDepResource = (
+  fromData: GitImportFormData | DeployImageFormData,
```
Comment (karthikjeeyar) [inline frontend/packages/knative-plugin/src/utils/create-knative-utils.ts]: nit: `fromData` to `formData`

File: frontend/packages/knative-plugin/src/utils/create-knative-utils.ts [hunk 2]
```diff
@@ -7,39 +7,27 @@ import {
...
+  const {
+    name,
+    application: { name: applicationName },
+    project: { name: namespace },
+    serverless: { scaling },
+    limits,
+    route: { unknownTargetPort },
+    labels,
+    image: { tag: imageTag },
+  } = fromData;
```
Comment (karthikjeeyar) [inline frontend/packages/knative-plugin/src/utils/create-knative-utils.ts]: replace it to `formData`

**Pull request merged**